main/src/lib.rs

```diff
@@ -44,7 +44,7 @@
 //! use rust_tokenizers::tokenizer::{BertTokenizer, Tokenizer, TruncationStrategy};
 //! use rust_tokenizers::vocab::{BertVocab, Vocab};
 //! let vocab_path = "path/to/vocab";
-//! let vocab = BertVocab::from_file(&vocab_path)?;
+//! let vocab = BertVocab::from_file(&vocab_path, Option::<&str>::None)?;
```

Comment:
It's a little bit messy during the WIP phase, but I give you my word, I'll make it look pretty before WIP-mark removal:)

We can add `from_file_with_special` or something like this and curry it in `from_file`, it remove turbofish syntax for most of the cases. For  rest of them we can make constant value:

```rust
const EMPTY_SPECIAL_MAP: Option<&str> = None;
```

I used `AsRef<Path>` instead of the `&str` because of two three reasons:
1. OS filepath is not necessary valid UTF8 string.
2. `Path` -> `&str` conversion force user to write extra error handling (you did it a lot in the tests).
3. `AsRef<Path>` is more acceptable with same level of correctness: you can use `&str`, `Path` and `PathBuf` with reference or without it.